src/libNode/DSBlockProcessing.cpp

```diff
@@ -218,24 +218,250 @@ void Node::LogReceivedDSBlockDetails([[gnu::unused]] const DSBlock& dsblock)
+    {
+        m_myShardMembersPubKeys.emplace_back(i.first);
+        m_myShardMembersNetworkInfo.emplace_back(i.second);
+
+        // Zero out my IP to avoid sending to myself
+        if (m_mediator.m_selfPeer == m_myShardMembersNetworkInfo.back())
+        {
+            m_consensusMyID = index; // Set my ID
+            m_myShardMembersNetworkInfo.back().m_listenPortHost = 0;
+        }
+
+        LOG_EPOCH(
+            INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+            " PubKey: "
+                << DataConversion::SerializableToHexStr(
+                       m_myShardMembersPubKeys.back())
+                << " IP: "
+                << m_myShardMembersNetworkInfo.back().GetPrintableIPAddress()
+                << " Port: "
+                << m_myShardMembersNetworkInfo.back().m_listenPortHost);
+
+        index++;
+    }
+
+#endif // IS_LOOKUP_NODE
+
+    return true;
+}
+
+void Node::LoadTxnSharingInfo(
+    [[gnu::unused]] const vector<unsigned char>& message,
+    [[gnu::unused]] unsigned int cur_offset)
+{
+#ifndef IS_LOOKUP_NODE
+
+    LOG_MARKER();
+
+    m_txnSharingIAmSender = false;
+    m_txnSharingIAmForwarder = false;
+    m_txnSharingAssignedNodes.clear();
+
+    vector<Peer> ds_receivers;
+    vector<vector<Peer>> shard_receivers;
+    vector<vector<Peer>> shard_senders;
+
+    TxnSharingAssignments::Deserialize(message, cur_offset, ds_receivers,
+                                       shard_receivers, shard_senders);
+
+    // m_txnSharingAssignedNodes below is basically just the combination of ds_receivers, shard_receivers, and shard_senders
+    // We will get rid of this inefficiency eventually
+
+    m_txnSharingAssignedNodes.emplace_back();
+
+    for (unsigned int i = 0; i < ds_receivers.size(); i++)
+    {
+        m_txnSharingAssignedNodes.back().emplace_back(ds_receivers.at(i));
+    }
+
+    for (unsigned int i = 0; i < shard_receivers.size(); i++)
+    {
+        m_txnSharingAssignedNodes.emplace_back();
+
+        for (unsigned int j = 0; j < shard_receivers.at(i).size(); j++)
+        {
+            m_txnSharingAssignedNodes.back().emplace_back(
+                shard_receivers.at(i).at(j));
+
+            if ((i == m_myShardID)
+                && (m_txnSharingAssignedNodes.back().back()
+                    == m_mediator.m_selfPeer))
+            {
+                m_txnSharingIAmForwarder = true;
+            }
+        }
+
+        m_txnSharingAssignedNodes.emplace_back();
+
+        for (unsigned int j = 0; j < shard_senders.at(i).size(); j++)
+        {
+            m_txnSharingAssignedNodes.back().emplace_back(
+                shard_senders.at(i).at(j));
+
+            if ((i == m_myShardID)
+                && (m_txnSharingAssignedNodes.back().back()
+                    == m_mediator.m_selfPeer))
+            {
+                m_txnSharingIAmSender = true;
+            }
+        }
+    }
+
+#endif // IS_LOOKUP_NODE
+}
+
+#ifndef IS_LOOKUP_NODE
+void Node::StartNewTxEpoch()
```

Comment:
A better name will be StartFirstTxEpoch()

Reply:
Done